hedera-node/docs/design/services/smart-contract-service/hedera-account-service-isAuthorized-design.md

```diff
@@ -0,0 +1,84 @@
+# HIP-632 (Hedera Account Service) `isAuthorized{Raw}` design
+
+This note addresses the implementation of `isAuthorized` and `isAuthorizedRaw` of 
+[HIP-632](https://hips.hedera.com/hip/hip-632) - "Hedera Account Service (HAS) 
+System Contract".
+
+## Preliminary notice w.r.t. "virtual addresses"
+
+HIP-632 as written refers a lot to 
+[HIP-631](https://hips.hedera.com/hip/hip-631) - "Account Virtual Addresses".
+However, that HIP, though `Accepted` will not be implemented.  (Or at least, not
+at any time soon.)  So HIP-632 needs to be read with that in mind.  Specifically, 
+anywhere that there's a provision for dealing with "virtual addresses", of which there
+may be several/many per Hedera account, there is in fact only _one_: the _alias_.
+
+* This mainly affects the method `getVirtualAddresses` which will return an array, but
+it will have _at most one_ EVM address in it.
+  
+## Phased implementation
+
+The first release of HIP-632 will include only `isAuthorizedRaw`.
+
+## Where does this go?
+
+A "hedera account service" system contract is being set up in
+`com.hedera.node.app.service.contract.impl.exec.systemcontracts.has`, and also at
+`...contract.impl.exec.processors.HasTranslatorsModule` (for the individual methods).
+
+Following the existing pattern for HAS each method will be given a package under 
+`...contract.impl.exec.systemcontracts.has` with two classes:
+* `[method]Translator` - which first confirms that the call is for the given method, and
+  then builds a `TransactionBody` for it.
+* `[method]Call` - which takes that `TransactionBody` and then has an `execute()` method
+  that does the work.
+  
+## Design for `isAuthorizedRaw`
+
+[TBD shortly as I fix up some problems with understanding HAS]
+
+`isAuthorizedRaw` has two flows: Either the argument address is a Hedera account, or it is
+an EVM address.
+
+* If an EVM address then this is simply the `ECRECOVER` precompile.  We have direct
```

Comment:
This may not be the case always right?
You could have an evm address that is a Hedera account id alias i.e. long zero.
In that case it wouldn't use `ecrecover` but would rather use the same flow as we could for an accountId

Reply:
As written HIP-632 specifically disallows long-zero addresses:

1) `isVirtualAddress` is defined to return `false` for a long-zero
2) When `isVirtualAddress` returns `false` then `isAuthorizedRaw` immediately returns `false`.

Maybe we could do better?

Reply:
Yeah, for the mentioned methods that makes sense but for `isAuthorized{Raw}()` given it just checks authorization I don't see any need for it not to work unless there's a concern unmentioned.

Reply:
Will add this suggestion (+ an update to the HIP) to the epic for the rest of -632.